scipy/signal/tests/test_signaltools.py

```diff
@@ -2525,29 +2525,49 @@ def test_negative_period_ValueError(self):
         assert_raises(ValueError, vectorstrength, events, period)
 
 
+def _tf2sos(b, a):
```

Comment:
why is this a good thing to add? there is a public `ts2sos` that's already imported, it seems preferable to use that instead?

Reply:
This requires #10972

Reply:
that's in now

Reply:
Simplified to a cast->-convert->cast function, with explanation (linalg does not support all of our tested dtypes)